src/main/java/org/openhab/tools/analysis/checkstyle/BuildPropertiesCheck.java

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.checkstyle;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+import org.eclipse.pde.core.build.IBuild;
+import org.eclipse.pde.core.build.IBuildEntry;
+import org.openhab.tools.analysis.checkstyle.api.AbstractStaticCheck;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+
+/**
+ * Checks if a build.properties file is valid.
+ *
+ * @author Petar Valchev - Initial implementation
+ * @author Svilen Valkanov - Changes in the logic for the source property
+ *
+ */
+public class BuildPropertiesCheck extends AbstractStaticCheck {
+    private static final String PROPERTIES_EXTENSTION = "properties";
+    private static final String BIN_INCLUDES_PROPERTY_NAME = "bin.includes";
+    private static final String OUTPUT_PROPERTY_NAME = "output..";
+    private static final String SOURCE_PROPERTY_NAME = "source..";
+
+    private static final String MISSING_PROPERTY_MSG = "Missing %s property in the %s file.";
+    private static final String MISSING_VALUE_MSG = "Property  %s in the %s file is missing value: ";
+
+    private static final String BUILD_PROPERTIES_FILE_NAME = "build." + PROPERTIES_EXTENSTION;
```

Comment:
Maybe beside the PROPERTIES_EXTENSTION also make this one shared? See other review.